bigtable/bigtable-proxy/src/main/java/com/google/cloud/bigtable/examples/proxy/channelpool/DataChannel.java

```diff
@@ -49,6 +49,10 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+/**
+ * Decorator for a Bigtable data plane connection to add channel warming via PingAndWarm. Channel
+ * warming will happen on creation and then every 3 minutes.
```

Comment:
does it happen on creation? The initial delay is also 3 minutes 🤔 ? `warmingExecutor.scheduleAtFixedRate(this::warmQuietly, 3, 3, TimeUnit.MINUTES);`

Reply:
nvm, I saw warm was called when the channel is created